src/main/java/ch/njol/skript/events/EvtEntityPotion.java

```diff
@@ -33,7 +33,7 @@ public class EvtEntityPotion extends SkriptEvent {
 
 	static {
 		Skript.registerEvent("Entity Potion Effect", EvtEntityPotion.class, EntityPotionEffectEvent.class,
-				"entity potion effect [modif[y|ication]] [[of] %potioneffecttypes%]")
+				"entity potion effect [modif[y|ication]] [[of] %potioneffecttypes%] [due to %entitypotioncause%]")
```

Comment:
you added the cause to the pattern, but it doesn't do anything. You need to add the logic to the event too.

Reply:
oh right yeah, i forgot lmao